File: ts-sdk/src/sdk/models/operations/postapireactions.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostApiReactionsCategoryEnum {
    Like = "like",
    Unicorn = "unicorn",
    ExplodingHead = "exploding-head",
    RaisedHands = "raised_hands",
    Fire = "fire"
}

export enum PostApiReactionsReactableTypeEnum {
    Comment = "Comment",
    Article = "Article",
    User = "User"
}


export class PostApiReactionsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=category" })
  category: PostApiReactionsCategoryEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reactable_id" })
  reactableId: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reactable_type" })
  reactableType: PostApiReactionsReactableTypeEnum;
}


export class PostApiReactionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostApiReactionsQueryParams;
}


export class PostApiReactionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: python-sdk/src/forem/models/operations/post_api_reactions.py
import dataclasses
from enum import Enum

class PostAPIReactionsCategoryEnum(str, Enum):
    LIKE = "like"
    READINGLIST = "readinglist"
    UNICORN = "unicorn"

class PostAPIReactionsReactableTypeEnum(str, Enum):
    COMMENT = "Comment"
    ARTICLE = "Article"
    USER = "User"


@dataclasses.dataclass
class PostAPIReactionsQueryParams:
    category: PostAPIReactionsCategoryEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'category', 'style': 'form', 'explode': True }})
    reactable_id: int = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_id', 'style': 'form', 'explode': True }})
    reactable_type: PostAPIReactionsReactableTypeEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_type', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class PostAPIReactionsRequest:
    query_params: PostAPIReactionsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class PostAPIReactionsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    
